components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -941,16 +943,86 @@ private void setTokenBindingToAccessTokenDO(AccessTokenDO dataDO, Connection con
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Persist all token to session mapping in the token binding table with binding type as "DEFAULT".
+     *
+     * @param sessionContextIdentifier SessionContextIdentifier
+     * @param tokenId                  TokenId.
+     * @param tenantId                 TenantId.
+     * @throws IdentityOAuth2Exception
+     */
+    public void storeTokenToSessionMapping(String sessionContextIdentifier, String tokenId, int tenantId)
+            throws IdentityOAuth2Exception {
+
+        if (isNotBlank(sessionContextIdentifier) && isNotBlank(tokenId)) {
+            Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+            try (PreparedStatement preparedStatement = connection.prepareStatement(STORE_TOKEN_BINDING)) {
+                preparedStatement.setString(1, tokenId);
+                preparedStatement.setString(2, DEFAULT_TOKEN_TO_SESSION_MAPPING);
+                preparedStatement.setString(3,
+                        OAuth2Util.getTokenBindingReference(sessionContextIdentifier));
+                preparedStatement.setString(4, sessionContextIdentifier);
+                preparedStatement.setInt(5, tenantId);
+                preparedStatement.execute();
+            } catch (SQLException e) {
+                String errorMsg = "Error while persisting token to session mapping for sessionId: " +
+                        sessionContextIdentifier;
+                if (log.isDebugEnabled()) {
+                    log.debug(errorMsg);
                 }
+                throw new IdentityOAuth2Exception(errorMsg, e);
+            } finally {
+                IdentityDatabaseUtil.closeConnection(connection);
             }
         }
     }
 
+    /**
+     * Get all tokens mapped to the bindingRef.
+     *
+     * @param sessionId SessionIdentifier
+     * @throws IdentityOAuth2Exception
+     * @return
+     */
+    public Set<String> getTokenIdBySessionIdentifier(String sessionId) throws IdentityOAuth2Exception {
+
+        String sql = SQLQueries.RETRIEVE_TOKENS_MAPPED_FOR_TOKEN_BINDING_VALUE;
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement prepStmt = null;
+        Set<String> tokenIds = new HashSet<>();
+        try {
+            prepStmt = connection.prepareStatement(sql);
+            prepStmt.setString(1, sessionId);
+            prepStmt.setString(2, DEFAULT_TOKEN_TO_SESSION_MAPPING);
+            try (ResultSet resultSet = prepStmt.executeQuery()) {
+                while (resultSet.next()) {
+                    tokenIds.add(resultSet.getString("TOKEN_ID"));
+                }
+            }
+
+        } catch (SQLException e) {
+            String errorMsg = "Error occurred while retrieving 'token id' for " +
+                    "binding value : " + sessionId;
+            throw new IdentityOAuth2Exception(errorMsg, e);
```

Comment:
need to handle unique constraint violation issue in the same manner as in the code level